### PR TITLE
add vcf_annotate_snpeff suborkflow and tests

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/vcf_annotate_snpeff/main.nf [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//
+// Run SNPEFF to annotate VCF files
+//
+
+include { SNPEFF           } from '../../../modules/nf-core/snpeff/main.nf'
+include { TABIX_BGZIPTABIX } from '../../../modules/nf-core/tabix/bgziptabix/main.nf'
+
+workflow VCF_ANNOTATE_SNPEFF {
+    take:
+    vcf          // channel: [ val(meta), vcf ]
```
Comment (matthdsm) [inline subworkflows/nf-core/vcf_annotate_snpeff/main.nf:10]: Just checking in, did we decide for or against a `ch_` prefix for channels inputs?

File: subworkflows/nf-core/vcf_annotate_snpeff/main.nf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    snpeff_db    //   value: db version to use
+    snpeff_cache //    path: /path/to/snpeff/cache (optionnal)
```
Comment (matthdsm) [inline subworkflows/nf-core/vcf_annotate_snpeff/main.nf:12]: `[meta2, snpeff_cache]`?

File: subworkflows/nf-core/vcf_annotate_snpeff/main.nf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    main:
+    ch_versions = Channel.empty()
+
+    SNPEFF(vcf, snpeff_db, snpeff_cache)
+    TABIX_BGZIPTABIX(SNPEFF.out.vcf)
```
Comment (matthdsm) [inline subworkflows/nf-core/vcf_annotate_snpeff/main.nf:18]: Personally, I'd vote for editing `snpeff` to make it output a compressed vcf + index, but that's just nitpicking

**Review by matthdsm — APPROVED** [review]
LGTM, just some general questions/comments

File: subworkflows/nf-core/vcf_annotate_snpeff/main.nf (hunk 3 above)
Comment (maxulysse) [reply subworkflows/nf-core/vcf_annotate_snpeff/main.nf:18]: My long term plan would be to actually scatter/gather the annotation, and gzip index the end VCF.

**Comment by matthdsm** [comment]
Oooh, that’s great! I’d like that for vep too!



Op 21 nov. 2022 om 21:34 heeft Maxime U. Garcia ***@***.***> het volgende geschreven:

﻿

@maxulysse commented on this pull request.

________________________________

In subworkflows/nf-core/vcf_annotate_snpeff/main.nf<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnf-core%2Fmodules%2Fpull%2F2535%23discussion_r1028492122&data=05%7C01%7CMatthias.DeSmet%40ugent.be%7C7c9ee73e0367493af35808dacbffc1e0%7Cd7811cdeecef496c8f91a1786241b99c%7C1%7C0%7C638046596584327001%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Vo8HR8KA8GIL8sPXySitC3tW4UALMnoBApIl%2F2pHCxY%3D&reserved=0>:

> +//
+
+include { SNPEFF           } from '../../../modules/nf-core/snpeff/main.nf'
+include { TABIX_BGZIPTABIX } from '../../../modules/nf-core/tabix/bgziptabix/main.nf'
+
+workflow VCF_ANNOTATE_SNPEFF {
+    take:
+    vcf          // channel: [ val(meta), vcf ]
+    snpeff_db    //   value: db version to use
+    snpeff_cache //    path: /path/to/snpeff/cache (optionnal)
+
+    main:
+    ch_versions = Channel.empty()
+
+    SNPEFF(vcf, snpeff_db, snpeff_cache)
+    TABIX_BGZIPTABIX(SNPEFF.out.vcf)


My long term plan would be to actually scatter/gather the annotation, and gzip index the end VCF.

File: subworkflows/nf-core/vcf_annotate_snpeff/main.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/nf-core/vcf_annotate_snpeff/main.nf:12]: no need for cache, it's the root path to folder

Comment (maxulysse) [reply subworkflows/nf-core/vcf_annotate_snpeff/main.nf:12]: cache is then resolved for each genome depending of the snpeff_db value

**Pull request merged**